app/Http/Controllers/UserController.php

```diff
@@ -101,14 +103,19 @@ public function getCourseDetails(Request $request) {
 	public function associate(Request $request) {
 		$user = Auth::user();
 		$courseKey = $request->courseKey;
-		$course = Course::where('course_key', $courseKey)->first();
+		$course = Course::with('teachers')->where('course_key', $courseKey)->first();
+
+		abort_if($course == null || $user == null, 509);
```

Comment:
509 Bandwidth Limit Exceeded ?

Reply:
Probably should send back an actual message to eh user that the course key is incorrect.